examples/fedpca_examples/dim_reduction/client.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+from pathlib import Path
+from typing import Tuple
+
+import flwr as fl
+import torch
+import torch.nn as nn
+from flwr.common import Config
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+from torchvision.transforms import transforms
+
+from examples.fedpca_examples.dim_reduction.mnist_model import MnistNet
+from fl4health.clients.basic_client import BasicClient
+from fl4health.preprocessing.pca_preprocessor import PCAPreprocessor
+from fl4health.utils.dataset import MNISTDataset
+from fl4health.utils.metrics import Accuracy
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+def get_mnist_dataset(data_dir: Path, train: bool) -> MNISTDataset:
+    transform = transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Normalize((0.5), (0.5)),
+        ]
+    )
+    return MNISTDataset(data_dir, train=train, transform=transform)
+
+
+class MnistFedPCAClient(BasicClient):
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        batch_size = self.narrow_config_type(config, "batch_size", int)
+        pca_path = Path(self.narrow_config_type(config, "pca_path", str))
+        new_dimension = self.narrow_config_type(config, "new_dimension", int)
+        pca_preprocessor = PCAPreprocessor(pca_path)
+        sampler = DirichletLabelBasedSampler(list(range(10)), sample_percentage=0.6, beta=0.75)
+
+        train_dataset = get_mnist_dataset(data_dir=self.data_path, train=True)
+
+        validation_dataset = get_mnist_dataset(data_dir=self.data_path, train=False)
+
+        train_loader = pca_preprocessor.reduce_dimension(
+            new_dimension=new_dimension, batch_size=batch_size, shuffle=True, sampler=sampler, dataset=train_dataset
+        )
+        val_loader = pca_preprocessor.reduce_dimension(
+            new_dimension=new_dimension,
+            batch_size=batch_size,
+            shuffle=False,
+            sampler=sampler,
+            dataset=validation_dataset,
+        )
+        return train_loader, val_loader
+
+    def get_criterion(self, config: Config) -> _Loss:
+        return torch.nn.CrossEntropyLoss()
+
+    def get_optimizer(self, config: Config) -> Optimizer:
+        return torch.optim.AdamW(self.model.parameters(), lr=0.0001)
+
+    def get_model(self, config: Config) -> nn.Module:
+        new_dimension = self.narrow_config_type(config, "new_dimension", int)
+        return MnistNet(input_dim=new_dimension).to(self.device)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Client Main")
+    parser.add_argument("--dataset_path", action="store", type=str, help="Path to the local dataset")
+    args = parser.parse_args()
+
+    DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    data_path = Path(args.dataset_path)
+
+    torch.manual_seed(47)
```

Comment:
I think it's probably best to use the functionality that Sana created in `from fl4health.utils.random import set_all_random_seeds` since it handles setting all of the relevant seeds for deterministic training? Perhaps you didn't do that for a reason. If so, a comment would be useful

Reply:
Yes that's a good point!